File: Snooper/src/androidTest/java/com/prateekj/snooper/SnooperInstrumentationRunner.java
package com.prateekj.snooper;

import android.app.Application;
import android.content.Context;
import androidx.test.runner.AndroidJUnitRunner;

public class SnooperInstrumentationRunner extends AndroidJUnitRunner {

  private Application application;

  @Override
  public Application newApplication(ClassLoader cl, String className, Context context) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
    return super.newApplication(cl, TestApplication.class.getName(), context);
  }

  @Override
  public void callApplicationOnCreate(Application app) {
    this.application = app;
    super.callApplicationOnCreate(app);
  }

  public Application getApplication() {
    return application;
  }
}

File: Snooper/src/androidTest/java/com/prateekj/snooper/SnooperInstrumentationRunner.kt
package com.prateekj.snooper

import android.app.Application
import android.content.Context
import androidx.test.runner.AndroidJUnitRunner

class SnooperInstrumentationRunner : AndroidJUnitRunner() {

  var application: Application? = null

  @Throws(
    InstantiationException::class,
    IllegalAccessException::class,
    ClassNotFoundException::class
  )
  override fun newApplication(cl: ClassLoader, className: String, context: Context): Application {
    return super.newApplication(cl, TestApplication::class.java.name, context)
  }

  override fun callApplicationOnCreate(app: Application) {
    this.application = app
    super.callApplicationOnCreate(app)
  }
}
